### ISSUE TITLE
tf.cond？？？？

### ISSUE THREAD

**Issue by melo19890618** [tensorflow/tensorflow#28658]
How to achieve in tensorflow: meet the condition, execute 1 (do not execute 2); do not satisfy the condition, execute 2 (do not execute 1)

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!

**Comment by melo19890618** [comment]
In the process of using the resnet50 for imagenet training, we used LARS to update the learning rate and calculate LR at each step of the training. The throughput of the training is about 5500. For this we intend to optimize and calculate the LR operation every few steps to improve throughput. In the original code, we perform compute_lr calculation every step.
I modified the code as shown below：
    gg is a tensor used to observe which step of training;
    2 is a constant, indicating that lr is calculated every 2 steps.

The code:

def compute_lr()
    coumpte_lr 
       ...
    stored_lr
       ...
    return lr
def get_larsvalue()
    get_stored_lr
       ...
    return lr

tf.cond(tf.cast(tf.equal(tf.mod(gg,2),0),tf.bool),lambda:self.compute_lr(),lambda: self.get_larsvalue())


But after I modified the code, the throughput dropped. After analysis, I think it is because tf.cond is not a lazy operation, it will execute both branches, which is obviously not what I want. I don't know how to code to complete my thoughts now, ask you to help. Thanks
